### PR TITLE
dcmtk 3.6.1 update

### PR DESCRIPTION
**Pull request by kkumler**
Added stable / devel targets, building against homebrew openssl, homebrew libiconv.
Is it desirable to include a way to select system/brewed libraries (at least for libiconv) for build?

Earlier PR and discussion: Homebrew/homebrew-versions#634

### CODE REVIEW

File: Library/Formula/dcmtk.rb [hunk 1]
```diff
@@ -12,39 +32,32 @@ class Dcmtk < Formula
...
-  # However, this patch can't be applied as-is, since it refers to
-  # some files that don't exist in the 3.6.0 release.
-  #
-  # This patch can be dropped once DCMTK makes a new release, but
-  # since this is a very rare occurrence (the last development preview
-  # release is from mid 2012), it seems justifiable to keep the patch
-  # ourselves for a while.
-  patch :DATA
+  depends_on "openssl" => :optional
+  depends_on "homebrew/dupes/libiconv" => :optional
```
Comment (mikemcquaid) [inline Library/Formula/dcmtk.rb]: What's the advantage of using this over the system version?

File: Library/Formula/dcmtk.rb [hunk 2]
```diff
@@ -12,39 +32,32 @@ class Dcmtk < Formula
...
   def install
     ENV.m64 if MacOS.prefer_64_bit?
 
     args = std_cmake_args
-    args << '-DDCMTK_WITH_DOXYGEN=YES' if build.with? "docs"
-    args << '-DDCMTK_WITH_OPENSSL=YES' if build.with? "openssl"
-    args << '..'
+    args << "-DDCMTK_WITH_DOXYGEN=YES" if build.with? "docs"
+    args << "-DDCMTK_WITH_OPENSSL=YES" if build.with? "openssl"
+    args << "-DWITH_OPENSSLINC=#{Formula["openssl"].opt_prefix}" if build.with? "openssl"
```
Comment (mikemcquaid) [inline Library/Formula/dcmtk.rb]: Share one `if` and `args` for the above two lines.

File: Library/Formula/dcmtk.rb [hunk 3]
```diff
@@ -12,39 +32,32 @@ class Dcmtk < Formula
...
+    args << "-DDCMTK_WITH_ICONV=YES" if build.with? "libiconv"
+    args << "-DLIBICONV_DIR=#{Formula["libiconv"].opt_prefix}" if build.with? "libiconv"
```
Comment (mikemcquaid) [inline Library/Formula/dcmtk.rb]: Similarly here.

File: Library/Formula/dcmtk.rb [hunk 4]
```diff
@@ -12,39 +32,32 @@ class Dcmtk < Formula
...
-    mkdir 'build' do
-      system 'cmake', *args
-      system 'make DOXYGEN' if build.with? "docs"
-      system 'make install'
+    mkdir "build" do
+      system "cmake", *args
+      system "make", "DOXYGEN" if build.with? "docs"
+      system "make", "install"
     end
   end
```
Comment (mikemcquaid) [inline Library/Formula/dcmtk.rb]: Can a test be added to do something more substantial than e.g. `--version` or `--help`? See `cmake.rb` for an example of an application formula with a good test and `tinyxml2.rb` for an example of a library formula with a good test. Thanks!

File: Library/Formula/dcmtk.rb (hunk 1 above)
Comment (kkumler) [inline Library/Formula/dcmtk.rb]: dcmtk and the system libiconv version had problems with dicom containing utf8.

Comment (mikemcquaid) [inline Library/Formula/dcmtk.rb]: Ok, probably want to declare an `option` to say what it does, thanks!

**Pull request closed without merging**